src/ConnectedMode/Migration/XmlFileCleaner.cs

```diff
@@ -73,12 +73,35 @@ public string Clean(string content, LegacySettings legacySettings, CancellationT
             nodesToRemove.ForEach(node =>
             {
                 Debug.Assert(node.ParentNode != null);
-                node.ParentNode.RemoveChild(node);
+
+                // If the only child elements in the parent element are white spaces and the node the whole parent elemnt can be removed.
+                var currentNode = ShouldRemoveParentElement(node) ? node.ParentNode : node;
```

Comment:
I am not sure currentNode makes much sense but I couldnt think of a better name.

Reply:
`nodeToRemove`? ;-)